tardis/montecarlo/src/cmontecarlo.c

```diff
@@ -234,6 +248,7 @@ compute_distance2line (rpacket_t * packet, const storage_model_t * storage)
           fprintf (stderr, "nu = %f\n", nu);
           fprintf (stderr, "doppler_factor = %f\n", doppler_factor);
           fprintf (stderr, "cur_zone_id = %" PRIi64 "\n", cur_zone_id);
+          rpacket_set_d_line (packet, NAN);
```

Comment:
My reasoning was, that since the calculation of `distance2line` fails we set the corresponding distance to NaN. Setting `d_boundary` in a function that calculates `d_line` seems counter intuitive to me.

The proper 'solution' would be to check the error of the distance calculations and terminate if something goes wrong.